### PR TITLE
Materialize adjoint(Givens) since Givens is immutable

### PR DESCRIPTION
**Pull request by andreasnoack**
This simplifies the code without any costs

### CODE REVIEW

File: stdlib/LinearAlgebra/src/givens.jl [hunk 1]
```diff
@@ -48,10 +48,13 @@ Rotation{T}(R::Rotation) where {T} = Rotation{T}([Givens{T}(g) for g in R.rotati
...
-adjoint(G::Givens) = Adjoint(G)
+adjoint(G::Givens) = Givens(G.i1, G.i2, G.c', -G.s)
 adjoint(R::Rotation) = Adjoint(R)
-Base.copy(aG::Adjoint{<:Any,<:Givens}) = (G = aG.parent; Givens(G.i1, G.i2, conj(G.c), -G.s))
-Base.copy(aR::Adjoint{<:Any,Rotation{T}}) where {T} = Rotation{T}(reverse!([copy(r') for r in aR.parent.rotations]))
+function Base.copy(aG::Adjoint{<:Any,<:Givens})
+    G = aG.parent
+    return Givens(G.i1, G.i2, conj(G.c), -G.s)
+end
+Base.copy(aR::Adjoint{<:Any,Rotation{T}}) where {T} = Rotation{T}(reverse!([r' for r in aR.parent.rotations]))
```
Comment (ararslan) [inline stdlib/LinearAlgebra/src/givens.jl]: Couldn't you do non-mutating `reverse(aR.parent.rotations)` instead of mutating a comprehension?

Comment (mbauman) [reply stdlib/LinearAlgebra/src/givens.jl]: You still need to map the adjoint operation in any case.

Comment (ararslan) [reply stdlib/LinearAlgebra/src/givens.jl]: Oh derp, I'm not sure how I missed that.

File: stdlib/LinearAlgebra/src/givens.jl [hunk 2]
```diff
@@ -339,17 +342,16 @@ function lmul!(G::Givens, A::AbstractVecOrMat)
...
+function rmul!(A::AbstractMatrix, G::Givens)
     @assert !has_offset_axes(A)
-    G = adjG.parent
     m, n = size(A, 1), size(A, 2)
     if G.i2 > n
         throw(DimensionMismatch("column indices for rotation are outside the matrix"))
     end
     @inbounds @simd for i = 1:m
         a1, a2 = A[i,G.i1], A[i,G.i2]
-        A[i,G.i1] =  a1*G.c + a2*conj(G.s)
```
Comment (mbauman) [inline stdlib/LinearAlgebra/src/givens.jl]: Strange that we used to take the `conj` of `G.s` here but not in the next line.  Was this an observable bug?

Comment (haampie) [reply stdlib/LinearAlgebra/src/givens.jl]: It's how the Given's rotation is defined, so should be correct

**Pull request merged**